spring-cloud-stream/src/main/java/org/springframework/cloud/stream/aggregate/NamespaceAwareSpringIntegrationMetricReader.java

```diff
@@ -0,0 +1,104 @@
+
+/**
+ * A customized version of {@link org.springframework.boot.actuate.metrics.integration.SpringIntegrationMetricReader} that
+ * provides support for customizing channels with a namespace prefix.
+ *
+ * @author Marius Bogoevici
+ * @see org.springframework.boot.actuate.metrics.integration.SpringIntegrationMetricReader for original implementation
+ */
+public class NamespaceAwareSpringIntegrationMetricReader implements MetricReader {
+
+	private final String namespace;
+
+	private final IntegrationMBeanExporter exporter;
+
+	public NamespaceAwareSpringIntegrationMetricReader(String namespace, IntegrationMBeanExporter exporter) {
+		Assert.hasText(namespace, "cannot be null or empty String");
+		Assert.notNull(exporter, "cannot be null");
+		this.namespace = namespace;
+		this.exporter = exporter;
+	}
+
+	@Override
+	public Metric<?> findOne(String metricName) {
+		return null;
+	}
+
+	@Override
+	public Iterable<Metric<?>> findAll() {
+		IntegrationMBeanExporter exporter = this.exporter;
+		List<Metric<?>> metrics = new ArrayList<Metric<?>>();
+		for (String name : exporter.getChannelNames()) {
+			String prefix = "integration.channel." + namespace + "." + name;
+			metrics.addAll(getStatistics(prefix + ".errorRate",
+					exporter.getChannelErrorRate(name)));
+			metrics.add(new Metric<Long>(prefix + ".sendCount",
+					exporter.getChannelSendCountLong(name)));
+			metrics.addAll(getStatistics(prefix + ".sendRate",
+					exporter.getChannelSendRate(name)));
+			metrics.add(new Metric<Long>(prefix + ".receiveCount",
+					exporter.getChannelReceiveCountLong(name)));
+		}
+		for (String name : exporter.getHandlerNames()) {
+			metrics.addAll(getStatistics("integration." + namespace + ".handler." + name + ".duration",
+					exporter.getHandlerDuration(name)));
+		}
+		metrics.add(new Metric<Integer>("integration." + namespace + ".activeHandlerCount",
+				exporter.getActiveHandlerCount()));
+		metrics.add(new Metric<Integer>("integration." + namespace + ".handlerCount",
+				exporter.getHandlerCount()));
+		metrics.add(new Metric<Integer>("integration." + namespace + ".channelCount",
+				exporter.getChannelCount()));
+		metrics.add(new Metric<Integer>("integration." + namespace + ".queuedMessageCount",
+				exporter.getQueuedMessageCount()));
+		return metrics;
+	}
+
+	private Collection<? extends Metric<?>> getStatistics(String name,
+			Statistics statistic) {
+		List<Metric<?>> metrics = new ArrayList<Metric<?>>();
+		metrics.add(new Metric<Double>(name + ".mean", statistic.getMean()));
+		metrics.add(new Metric<Double>(name + ".max", statistic.getMax()));
+		metrics.add(new Metric<Double>(name + ".min", statistic.getMin()));
+		metrics.add(
+				new Metric<Double>(name + ".stdev", statistic.getStandardDeviation()));
+		metrics.add(new Metric<Long>(name + ".count", statistic.getCountLong()));
+		return metrics;
+	}
+
+	@Override
+	public long count() {
+		int totalChannelCount = this.exporter.getChannelCount() * 11;
```

Comment:
The original code is here: https://github.com/spring-projects/spring-boot/commit/5ceb35473d656e086aa13fbc433df003377a4819#diff-d592286eb2093acb9f86af6d5d0dcdc8R78.

I think there is some shift logic to distinguish different counts in a single value.
Or just classical @dsyer 's magic :smile: